src/core/mapping/base_mapper.cc

```diff
@@ -165,12 +165,9 @@ void BaseMapper::select_task_options(const Legion::Mapping::MapperContext ctx,
     LEGATE_ABORT;
   }
 
-  auto target      = legate_mapper_->task_target(legate_task, options);
-  auto local_range = machine.slice(target, machine_desc);
-#ifdef DEBUG_LEGATE
-  assert(!local_range.empty());
-#endif
-  output.initial_proc = local_range.first();
+  auto target = legate_mapper_->task_target(legate_task, options);
+  // The initial processor just needs to have the same kind as the eventual target of this task
+  output.initial_proc = machine.procs(target).front();
```

Comment:
Is there a scenario in which even this might be empty?

Reply:
no it can't. that should have already been caught at this line: https://github.com/nv-legate/legate.core/blob/branch-23.05/legate/core/context.py#L177